### PR TITLE
json to xml

### PR DESCRIPTION
**Pull request by qadan**

### CODE REVIEW

File: src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.Authenticator;
+import java.net.PasswordAuthentication;
+import java.util.Scanner;
+
+/**
+ * Authenticator for connecting to Fedora.
+ */
+class FedoraAuthenticator extends Authenticator {
```
Comment (qadan) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: Imma move it down into the public class

File: src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+   * @param fedoraUser
+   *   The Fedora username to connect with.
+   * @param fedoraPass
+   *   The Fedora password to connect with.
+   *
+   * @return InputStream
+   *   An InputStream at the constructed dissemination point.
+   */
+  public static InputStream getDatastreamDisseminationInputStream(String pid, String dsId, String fedoraBase, String fedoraUser, String fedoraPass) {
+    URL url;
```
Comment (qadan) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: I think Eclipse did that and I forgot to fix it. Imma move the definition down

File: src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+   * @return String
+   *   The text of the given datastream.
+   */
+  public static String getRawDatastreamDissemination(String pid, String dsId, String fedoraBase, String fedoraUser, String fedoraPass) {
+    try {
+      URL url = getDatastreamDisseminationURL(pid, dsId, fedoraBase, fedoraUser, fedoraPass);
+      InputStream dsStream = url.openStream();
+      Scanner scanner = new Scanner(dsStream);
+      // If no content, return an empty string.
+      String dsString = scanner.hasNext() ? scanner.next() : "";
```
Comment (qadan) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: I'm thinking the second option sounds better

Comment (qadan) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: Also, will add the 'start of string' delimiter ... was using JSON that has no CR/LF so it never came up, but this is more generic than that

File: src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      logger.debug(String.format("getRawDatastreamDissemination (pid: %s, DSID: %s): %s", pid, dsId, dsString));
```
Comment (qadan) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: Should the `logger.warn()` calls that use `String.format()` similarly incorporate `logger.isWarnEnabled()`?

Comment (adam-vessey) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: Given how logging levels would typically be "info" (between the two) or "warning" proper... I don't think it really matters.

File: src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  /**
+   * Turns some parameters into a datastream dissemination URL.
+   *
+   * @return URL
+   *   A URL object using the given parameters.
+   */
+  protected static final URL getDatastreamDisseminationURL(String pid, String dsId, String fedoraBase, String fedoraUser, String fedoraPass) throws MalformedURLException {
+    // Set the authenticator.
+    FedoraAuthenticator auth = new FedoraAuthenticator();
+    auth.setCredentials(fedoraUser, fedoraPass);
```
Comment (qadan) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: Will do so. Also realizing that this authenticator-setting chunk is sort of unexpected behaviour for something that claims to simply get a URL, so will move it up to the dissemination getter

Comment (adam-vessey) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: Will it use it if it's present in the URL proper? Like... the whole "http://user:pass@host/whatever" business?

Comment (qadan) [inline src/main/java/ca/discoverygarden/gsearch_extensions/FedoraUtils.java]: Tested that when t isn't present in the URL proper when using the authenticator (`url.getUserInfo()` is empty)

**Pull request merged**